hazelcast/src/main/java/com/hazelcast/internal/networking/ChannelHandler.java

```diff
@@ -0,0 +1,139 @@
+     * should  modify this field.
+     */
+    protected Channel channel;
+
+    protected S src;
+    protected D dst;
+
+    /**
+     * Gets the source of this ChannelHandler.
+     *
+     * This method should only be called from the thread that owns this
+     * ChannelHandler.
+     *
+     * @return the source; could be null.
+     */
+    public S src() {
+        return src;
+    }
+
+    /**
+     * Sets the source of this ChannelHandler.
+     *
+     * This method should only be called from the thread that owns this
+     * ChannelHandler.
+     *
+     * @param src the new source; is allowed to be null.
+     */
+    public void src(S src) {
+        this.src = src;
+    }
+
+    /**
+     * Gets the destination of this ChannelHandler.
+     *
+     * This method should only be called from the thread that owns this
+     * ChannelHandler.
+     *
+     * @return the destination; could be null.
+     */
+    public D dst() {
+        return dst;
+    }
+
+    /**
+     * Sets the destination of this ChannelHandler.
+     *
+     * This method should only be called from the thread that owns this
+     * ChannelHandler.
+     *
+     * @param dst the new destination; is allowed to be null.
+     */
+    public void dst(D dst) {
+        this.dst = dst;
+    }
+
+    /**
+     * Sets the Channel.
+     *
+     * Should only be called by the {@link ChannelInboundPipeline}.
+     *
+     * @param channel the Channel
+     * @return this instance
+     */
+    public final H setChannel(Channel channel) {
+        this.channel = channel;
+        return (H) this;
+    }
+
+    /**
+     * Gets called when this ChannelHandler is added to the pipeline.
+     *
+     * This method should execute very quickly because it could be executed on
+     * an io thread.
+     *
+     * Will be called from a thread owning this handler.
+     */
+    public void handlerAdded() {
+    }
+
+    /**
+     * Intercepts an error that is thrown while processing the inbound or
+     * outbound pipeline.
+     *
+     * The default implementation doesn't do anything and the original
+     * {@link Throwable} is send to the {@link ChannelErrorHandler}.
+     *
+     * However in some cases like with TLS a more informative exception
+     * needs to be thrown. To prevent loosing stacktraces, it is best to
+     * include the original Throwable as cause in the new Throwable.
+     *
+     * This method will only be called from the thread owning the pipeline
+     * this handler is part of.
+     *
+     * No modifications should be done to the pipeline and no attempts
+     * should be made to 'repair' the channel. When a Throwable is thrown,
+     * it is the end of the channel.
+     *
+     * @param error the Throwable thrown while processing the pipeline.
+     * @throws Throwable the new Throwable
+     */
+    public void interceptError(Throwable error) throws Throwable {
+    }
+
+    public void channelInactive() throws Exception {
```

Comment:
Can you add a few words here? I don't find `channelInactive()` self explaining.

Reply:
Good point. This is actually the last point that needs to be addressed; closing the channel so that the SSLEngine is closed correctly.

So in other words; this part isn't complete yet.